gui/wxpython/gui_core/pyedit.py

```diff
@@ -400,6 +412,28 @@ def OnOpen(self, event):
         """Handle open event but ask about replacing content first"""
         if self.CanReplaceContent('file'):
             self.Open()
+            self.recent_files.AddFileToHistory(
+                filename=self.filename,
+            )
+
+    def OpenRecentFile(self, path, file_exists):
+        """Try open recent file and read content
+
+        :param str path: file path
+        :param bool file_exists: file path exists
+
+        :return: None
+        """
+        if self.CanReplaceContent(by_message='file'):
+            if not file_exists:
+                GError(
+                    _("File path <{}> doesn't exist.".format(path)),
+                    parent=self.guiparent,
+                )
+            else:
+                self.filename = path
```

Comment:
Don't you need to set tempfile to false like in Open? (I don't know.) Perhaps worth a dedicated function for the file opening itself. That might good, e.g., for adding proper error handling when opening the file (as opposed to only having checks for common problems beforehand).

Reply:
I made two private methods for handling open file [_openFile()](https://github.com/OSGeo/grass/pull/1021/commits/93c6d67962a30a205c4a954a105d9757242ef05e#diff-3bb13543851c2a00667fabbae7b2c76b04e4610bf338784944c13e82fc471f29R280) method and for write file [_writeFile()](https://github.com/OSGeo/grass/pull/1021/commits/93c6d67962a30a205c4a954a105d9757242ef05e#diff-3bb13543851c2a00667fabbae7b2c76b04e4610bf338784944c13e82fc471f29R306) method.